### PR TITLE
Add object store interface to API client + tests 

### PR DESCRIPTION
**Pull request by sarahwooders**
* Implemented some basic object store interfacing in the API client for convenience. 
* Fixed bug with deleted buckets (need to delete keys first) for AWS/GCP 
* Added tests to create/delete buckets and upload/download objects for AWS/GCP 

I wouldn't get Azure permissions to work, so made an issue #774.

### CODE REVIEW

File: skyplane/obj_store/s3_interface.py [hunk 1]
```diff
@@ -79,6 +80,18 @@ def create_bucket(self, aws_region):
...
     def delete_bucket(self):
+        # delete 1000 keys at a time
+        keys = []
+        for key in self.list_objects():
+            keys.append(key.key)
+            if len(keys) == 1000:
+                self.delete_objects(keys)
+                keys = []
+                print("Deleted 1000 keys")
+        self.delete_objects(keys)
```
Comment (zizhong) [inline skyplane/obj_store/s3_interface.py:91]: Seems that the `delete_objects` already has the logic of taking 1000 keys as the step to iterate.

File: skyplane/api/client.py [hunk 2]
```diff
@@ -142,3 +142,48 @@ def dataplane(
             return Dataplane(clientid=self.clientid, topology=topo, provisioner=self.provisioner, transfer_config=self.transfer_config)
         else:
             raise NotImplementedError(f"Dataplane type {solver_type} not implemented")
+
+    def download_object(self, bucket_name: str, provider: str, key: str, filename: str):
```
Comment (parasj) [inline skyplane/api/client.py:146]: Can we move this to another class and add another top-level API primitive? This seem separate from the client itself. You can call it via:
```
obj_store = client.object_store()
```

File: skyplane/api/client.py [hunk 3]
```diff
@@ -142,3 +142,48 @@ def dataplane(
...
+        obj_store = ObjectStoreInterface.create(f"{provider}:infer", bucket_name)
+        obj_store.upload_object(filename, key)
+
+    def exists(self, bucket_name: str, provider: str, key: str) -> bool:
+        obj_store = ObjectStoreInterface.create(f"{provider}:infer", bucket_name)
+        return obj_store.exists(key)
+
+    def bucket_exists(self, bucket_name: str, provider: str) -> bool:
+        # azure not implemented
+        if provider == "azure":
```
Comment (parasj) [inline skyplane/api/client.py:160]: Is there a reason we don't have Azure support yet? Would this design generalize?

File: skyplane/obj_store/s3_interface.py (hunk 1 above)
Comment (parasj) [reply skyplane/obj_store/s3_interface.py:91]: It's maybe better to use a generator here:
```python
    def delete_bucket(self):
        for batch in batch_generator(self.list_objects(), 1000):
            self.delete_objects([obj.key for obj in batch])
        # actually remove the bucket
```


There's already a convenience function that will batch data from a generator:
https://github.com/skyplane-project/skyplane/blob/da90f82602eda08f06ca32336ba10c7f1c7774a7/skyplane/api/transfer_job.py#L286-L299

**Review by parasj — APPROVED** [review]
LGTM!

**Pull request merged**